src/components/KanaQuiz/Question.tsx

```diff
@@ -0,0 +1,75 @@
+import styled from 'styled-components';
+import { KanaType } from '../../types/kana';
+import { CurrentQuestion } from '../../types/kanaQuiz';
+
+type QuestionProps = {
+  currentQuestion: CurrentQuestion;
+  onChange: (answer: KanaType) => void;
+};
+
+export default function Question({ currentQuestion, onChange }: QuestionProps) {
+  return (
+    <QuestionStyled>
+      <legend>{currentQuestion.question?.kana}</legend>
+      {currentQuestion.answers.map(answer => (
+        <label key={answer?.id}>
```

Comment:
Is optional chaining here necessary? Are there cases where `answer` might be undefined or null? As `currentQuestion.answers` should be an array of `KanaType`, it should either be empty or have answers. If it's empty, the loop won't be run, if it has answers, they should all have an id. Only if for some reason the array contains other values than correct `answers: KanaType[]` will the optional chaining be needed.

Same for the optional chaining above and below. The way the types are defined, if the code reaches this point, optional chaining should never be necessary.

Whenever I see optional chaining, I assume that there are cases where they might be needed. But either these cases do not exist in your code, or they are not reflected in the types.